### ISSUE TITLE
Unable to cast object of type 'Microsoft.VisualStudio.CommonIDE.Solutions.SolutionPropertyBag' to type 'Ankh.Scc.Native.ICOMPropertyBag'

### ISSUE THREAD

**Issue by Rand-Random** [AmpScm/AnkhSVN#44]
**Edit:**
Since the build artifacts are expired, here a version I downloaded some time ago

[Ankh-VSIX-2022.zip](https://github.com/AmpScm/AnkhSVN/files/15106354/Ankh-VSIX-2022.zip)

This version should fix the below mentioned issue.

Version of the provided file:
2.9.120.0
2.9.120-d119c2b3ac9e09a09631eb7801f7450ce5d41c3b

Here a screenshot of the content, after I renamed the containing *.vsix file to *.zip and extraced its content:
![image](https://github.com/user-attachments/assets/ce48a66a-40a0-4388-9d40-ba8ca2b5eaf7)

___

Upon opening one of many solutions I am getting this error.
The error happens everytime I open this one solution, other solutions can be opened without an error.

Except the annoyance of closing the message box everytime I open the solution, I don't see any other issues.
Everything seems to work eg. commiting/updating and everything else ankh has to offer, just this error message upon opening the solution, nothing more nothing less.

> Unable to cast object of type 'Microsoft.VisualStudio.CommonID.Solutions.SolutionPropertyBag' to type 'Ankh.Scc.Native.ICOMPropertyBag'

> System.InvalidCastException: Unable to cast object of type 'Microsoft.VisualStudio.CommonIDE.Solutions.SolutionPropertyBag' to type 'Ankh.Scc.Native.ICOMPropertyBag'.
   at Ankh.VSPackage.AnkhSvnPackage.ReadSolutionProps(IVsHierarchy pHierarchy, String pszProjectName, String pszProjectMk, String pszKey, Int32 fPreLoad, IPropertyBag pPropBag)

![image](https://user-images.githubusercontent.com/9868626/164229767-cf421eb0-3cee-486d-9434-da5b4c32bf15.png)

```
os-version=10.0.22000.0 
vs-version=17.0.32408.312 
svn-version=1.14.1 
sharpsvn-version=1.14001.54.0 
ankh-version=2.9.89.0 
```

**Comment by rhuijben** [comment]
Interesting issue. Looks like the code in vs was converted to managed without proper com compatibility... We need a fix there.

**Comment by Rand-Random** [comment]
Thanks for the very fast fix.
I wonder is there some automatic build where I can grab a version with this commit?

**Comment by rhuijben** [comment]
If you click on Actions in the project and look at the last successful build, you should see artifacts that you can download... These are vsix packages that can be installed in VS

**Comment by andrew-tyas-idox** [comment]
Still an issue in 17.4.3

**Comment by Rand-Random** [comment]
@andrew-tyas-idox 
You installed the latest build of the plugin, as was mentioned in this topic?
You may find the currently latest version here: https://github.com/AmpScm/AnkhSVN/actions/runs/3633025147

**Comment by andrew-tyas-idox** [comment]
> @andrew-tyas-idox You installed the latest build of the plugin, as was mentioned in this topic? You may find the currently latest version here: https://github.com/AmpScm/AnkhSVN/actions/runs/3633025147

Yes, that resolves it. I had installed the latest Release version instead,

**Comment by martin-belcher** [comment]
> > @andrew-tyas-idox You installed the latest build of the plugin, as was mentioned in this topic? You may find the currently latest version here: https://github.com/AmpScm/AnkhSVN/actions/runs/3633025147
> 
> Yes, that resolves it. I had installed the latest Release version instead,

TIP: You have to sign up to GitHub and be logged in otherwise the build Artifacts are not downloadable.

**Comment by re14nz** [comment]
The ankhsvn package is expired. 
Where can I download the latest update?
Thank you.

**Comment by Rand-Random** [comment]
@re14nz 
Click on Actions at the top, navigate to the latest Build Action, ans at the bottom you will find the Assets the build produced.

**Comment by ChumaniCodeRR** [comment]
Hi,

I have a similar problem, we currently migrating to Visual Studio 2022. From Visual Studio 2012

How can I fix the problem with the Ankh sub-version? 
![errorANkh](https://github.com/AmpScm/AnkhSVN/assets/50732882/bfa23de3-dd52-4473-9bc0-a498c52374c3)

**Comment by Rand-Random** [comment]
@ChumaniCodeRR 
Did you actually read, the messages here?
solution was already posted a couple of times.

**Comment by JesperOlsen** [comment]
There are no successful builds where the artifacts are not expired

**Comment by Rand-Random** [comment]
Not sure why I deserve the thumbs up, isn't like I am doing the builds/artifacts, but okay you do you.

Here a version I downloaded some time ago, give it a try:
[Ankh-VSIX-2022.zip](https://github.com/AmpScm/AnkhSVN/files/15106308/Ankh-VSIX-2022.zip)

**Comment by vman2000** [comment]
> There are no successful builds where the artifacts are not expired

Yes, I couldn't find any that worked. In the end I just downloaded the code a built the solution, and picked the Ankh.Package.vsix out of src\Ankh.Package\bin.

That was quicker than looking for a build and solved the problem perfectly.

**Comment by Rand-Random** [comment]
@vman2000 
the attached file at the first post didn’t work?

**Comment by vman2000** [comment]
> @vman2000 the attached file at the first post didn’t work?

Honestly, whilst I appreciate you are trying to help and I'm sure its okay, a "here's a version I downloaded some time ago" zip with no version/build info from some guy I dont know on a two year old comment didn't fill me with the same confidence as a quick download and build of the current source.

Its only one extra click to build the latest commit from a known legit source.

**Comment by Rand-Random** [comment]
@vman2000 
Just to let you know, doesn't help you and may not increase the confidence but maybe it helps others.
Edited the first post and included the version number + a screenshot of its content.
